test/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoTest.java

```diff
@@ -0,0 +1,194 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import javax.annotation.Resource;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.TestUtils;
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.models.ClientInfo;
+import org.sagebionetworks.bridge.models.schedules.ScheduleContext;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifierImpl;
+
+import com.google.common.collect.Sets;
+
+@ContextConfiguration("classpath:test-context.xml")
+@RunWith(SpringJUnit4ClassRunner.class)
+public class DynamoSubpopulationDaoTest {
+    
+    private static final String SUBPOP_1 = "Subpop 1";
+    private static final String SUBPOP_2 = "Subpop 2";
+    private static final String SUBPOP_3 = "Subpop 3";
+    private static final String SUBPOP_4 = "Subpop 4";
+
+    StudyIdentifier studyId;
+    
+    @Resource
+    DynamoSubpopulationDao dao;
+    
+    @BeforeClass
+    public static void beforeClass() {
+        DynamoInitializer.init(DynamoSubpopulation.class);
+    }
+    
+    @Before
+    public void before() {
+        studyId = new StudyIdentifierImpl(TestUtils.randomName(DynamoSubpopulationDaoTest.class));
+    }
+    
+    @After
+    public void after() {
+        dao.deleteAllSubpopulations(studyId);
+        assertTrue(dao.getSubpopulations(studyId, true).isEmpty());
+    }
+    
+    @Test
+    public void crudSubpopulationOK() {
+        DynamoSubpopulation subpop = new DynamoSubpopulation();
+        subpop.setStudyIdentifier(studyId.getIdentifier());
+        subpop.setName("Name");
+        subpop.setDescription("Description");
+        subpop.setMinAppVersion(2);
+        subpop.setMaxAppVersion(10);
+        
+        // CREATE
+        Subpopulation savedSubpop = dao.createSubpopulation(subpop);
+        assertNotNull(savedSubpop.getGuid());
+        assertNotNull(savedSubpop.getVersion());
+        
+        // READ
+        Subpopulation retrievedSubpop = dao.getSubpopulation(studyId, savedSubpop.getGuid());
+        assertEquals(savedSubpop, retrievedSubpop);
+        
+        // UPDATE
+        retrievedSubpop.setName("Name 2");
+        retrievedSubpop.setDescription("Description 2");
+        Subpopulation finalSubpop = dao.updateSubpopulation(retrievedSubpop);
```

Comment:
Do we want to set deleted=false to make sure the DAO enforces not being able to delete through an update?